File: BorrowService/Models/Loan.cs
namespace BorrowService.Models; // Put entity in BorrowService.Models namespace.

public class Loan // Represents a borrow transaction.
{
    public int Id { get; set; } // Primary key.

    public int UserId { get; set; } // User id from JWT/auth service.
    public int BookId { get; set; } // Book id from catalog service.
    public DateTime BorrowedAtUtc { get; set; } // Borrow timestamp.
    public DateTime DueAtUtc { get; set; } // Due date.
    public DateTime? ReturnedAtUtc { get; set; } // Return timestamp if returned.
}

File: elib_mobile_new/lib/models/loan.dart
class Loan {
  final int id;
  final int bookId;
  final String borrowedAtUtc;
  final String dueAtUtc;
  final String? returnedAtUtc;

  Loan({
    required this.id,
    required this.bookId,
    required this.borrowedAtUtc,
    required this.dueAtUtc,
    this.returnedAtUtc,
  });

  factory Loan.fromJson(Map<String, dynamic> json) {
    return Loan(
      id: json['id'] as int,
      bookId: json['bookId'] as int,
      borrowedAtUtc: json['borrowedAtUtc']?.toString() ?? '',
      dueAtUtc: json['dueAtUtc']?.toString() ?? '',
      returnedAtUtc: json['returnedAtUtc']?.toString(),
    );
  }
}